src/pages/SwapPage/Swap.tsx

```diff
@@ -38,6 +58,67 @@ function Swap({ swap }: SwapProps) {
   const [alphaTradeAction, betaTradeAction] =
     properties.role === Role.Alice ? tradeActions : tradeActions.reverse();
 
+  const communicationActions = (swap.actions || []).filter(
+    action => action.name === "accept" || action.name === "decline"
+  );
+
+  const ledgerActions = (swap.actions || []).filter(
+    action => action.name !== "accept" && action.name !== "decline"
+  );
+  const alphaLedgerActions =
```

Comment:
📣 There seems to be some logic here that could be tested. Please add some UI tests to the component that verifies the workflow for the different roles.

Reply:
I agree. I think in general the swap page component needs UI tests. I've opened #92 to track this.